src/Codec/Xlsx/StyleSheet.hs

```diff
@@ -0,0 +1,1123 @@
+    , elementAttributes = Map.empty
+    , elementNodes      = map NodeElement $ [
+         -- TODO: numFmts
+         elementList "fonts"   $ map (toElement "font")   _styleSheetFonts
+       , elementList "fills"   $ map (toElement "fill")   _styleSheetFills
+       , elementList "borders" $ map (toElement "border") _styleSheetBorders
+         -- TODO: cellStyleXfs
+       , elementList "cellXfs" $ map (toElement "xf")     _styleSheetCellXfs
+         -- TODO: cellStyles
+         -- TODO: dxfs
+         -- TODO: tableStyles
+         -- TODO: colors
+         -- TODO: extLst
+       ]
+    }
+
+-- | See @CT_Xf@, p. 4486
+instance ToElement CellXf where
+  toElement nm CellXf{..} = Element {
+      elementName       = nm
+    , elementNodes      = map NodeElement . catMaybes $ [
+          toElement "alignment"  <$> _cellXfAlignment
+        , toElement "protection" <$> _cellXfProtection
+          -- TODO: extLst
+        ]
+    , elementAttributes = Map.fromList . catMaybes $ [
+          "numFmtId"          .=? _cellXfNumFmtId
+        , "fontId"            .=? _cellXfFontId
+        , "fillId"            .=? _cellXfFillId
+        , "borderId"          .=? _cellXfBorderId
+        , "xfId"              .=? _cellXfId
+        , "quotePrefix"       .=? _cellXfQuotePrefix
+        , "pivotButton"       .=? _cellXfPivotButton
+        , "applyNumberFormat" .=? _cellXfApplyNumberFormat
+        , "applyFont"         .=? _cellXfApplyFont
+        , "applyFill"         .=? _cellXfApplyFill
+        , "applyBorder"       .=? _cellXfApplyBorder
+        , "applyAlignment"    .=? _cellXfApplyAlignment
+        , "applyProtection"   .=? _cellXfApplyProtection
+        ]
+    }
+
+-- | See @CT_CellAlignment@, p. 4482
+instance ToElement Alignment where
+  toElement nm Alignment{..} = Element {
+      elementName       = nm
+    , elementNodes      = []
+    , elementAttributes = Map.fromList . catMaybes $ [
+          "horizontal"      .=? _alignmentHorizontal
+        , "vertical"        .=? _alignmentVertical
+        , "textRotation"    .=? _alignmentTextRotation
+        , "wrapText"        .=? _alignmentWrapText
+        , "relativeIndent"  .=? _alignmentRelativeIndent
+        , "indent"          .=? _alignmentIndent
+        , "justifyLastLine" .=? _alignmentJustifyLastLine
+        , "shrinkToFit"     .=? _alignmentShrinkToFit
+        , "readingOrder"    .=? _alignmentReadingOrder
+        ]
+    }
+
+-- | See @CT_Border@, p. 4483
+instance ToElement Border where
+  toElement nm Border{..} = Element {
+      elementName       = nm
+    , elementAttributes = Map.fromList . catMaybes $ [
+          "diagonalUp"   .=? _borderDiagonalUp
+        , "diagonalDown" .=? _borderDiagonalDown
+        , "outline"      .=? _borderOutline
+        ]
+    , elementNodes      = map NodeElement . catMaybes $ [
+          toElement "start"      <$> _borderStart
+          -- Using position of @start@ for the not-per-spec @left@ element
```

Comment:
So it appears that actually minor code change needs to be done, as I've shown you in one of my previous comments, the order is start,end,left,right... The spec talks about left/right in the text and preset definitions have them but for some reason schemas lack their definitions. What would you say if would put the correct order of elements here in that PR about spec references?